### ISSUE TITLE
wsmancli failed to build because of a broken/incomplete openwsman build/bottle

### ISSUE THREAD

**Issue by apjanke** [Homebrew/legacy-homebrew#44047]
https://gist.github.com/c5035762e9d1c5bc78d0

Missing header file.

Looks like it might be an issue in `openwsman` itself? This missing header is included from another `openwsman` header.

```
clang -DHAVE_CONFIG_H -I. -I..    -g -O2 		-Wall -Wstrict-prototypes -Wmissing-declarations 		-Wmissing-prototypes -Wnested-externs -Wpointer-arith                 -Wunused -Werror -I/usr/local/Cellar/openwsman/2.6.0_1/include/openwsman  -DPACKAGE_BUILDTS=\"201509161611\" -c -o wseventmgr.o wseventmgr.c
In file included from wseventmgr.c:51:
In file included from /usr/local/Cellar/openwsman/2.6.0_1/include/openwsman/wsman-client-api.h:47:
/usr/local/Cellar/openwsman/2.6.0_1/include/openwsman/wsman-epr.h:44:10: fatal error: 'wsman-key-value.h' file not found
#include "wsman-key-value.h"
         ^
In file included from wsman.c:51:
In file included from /usr/local/Cellar/openwsman/2.6.0_1/include/openwsman/wsman-client.h:48:
In file included from /usr/local/Cellar/openwsman/2.6.0_1/include/openwsman/wsman-client-api.h:47:
/usr/local/Cellar/openwsman/2.6.0_1/include/openwsman/wsman-epr.h:44:10: fatal error: 'wsman-key-value.h' file not found
#include "wsman-key-value.h"
         ^
1 error generated.
make[1]: *** [wseventmgr.o] Error 1
```

```
[/usr/local/Cellar/openwsman/2.6.0_1 on ⇄ master]
$ find . -name wsman-key-value.h
[/usr/local/Cellar/openwsman/2.6.0_1 on ⇄ master]
$
```

**Comment by apjanke** [comment]
Same error for me on OS X 10.9.5 with Xcode 6.2. Not 10.11-specific.

**Comment by UniqMartin** [comment]
The problem here seems to be that our formula for `openwsman` still uses the `autoconf`/`automake` build system, but upstream since has switched to CMake. A good indication of this is the age of `Makefile.am` vs. their `CMakeLists.txt` counterparts (the former have not been touched in years), e.g. as can be seen in https://github.com/Openwsman/openwsman/tree/v2.6.2/include.

This explains why the missing header `wsman-key-value.h` is in the source tarball but doesn't get installed (and is thus also missing in the bottle). Switching to CMake as the build system for `openwsman` should fix this. (Too bad upstream stopped maintaining the `autoconf`/`automake` build system without removing it.)

**Comment by ilovezfs** [comment]
Upstream seems to have fixed this. I cannot reproduce the issue.

**Comment by UniqMartin** [comment]
Just wanted to mention that I tried (and failed) moving the `openwsman` formula to CMake. The `CMakeLists.txt` files used by that project make so many Linux-specific assumptions that I gave up working around them. So the situation seems to be:

- The Autotools build system [is deprecated by upstream](https://github.com/Openwsman/openwsman/wiki/Build-requirements), hasn't been updated in years, and produces semi-broken builds, but is still in use in our formula.
- The CMake build system is maintained, but apparently wasn't tested on anything other than Linux.

Maybe the “solution” is to just boneyard `wsmancli`, `openwsman`, and `sblim-sfcc`? It seems like those formulae haven't seen much use and much love from our contributors.

**Comment by mikemcquaid** [comment]
> The CMake build system is maintained, but apparently wasn't tested on anything other than Linux.

Sounds like it's worth asking upstream if they aim to support OS X. If not: 💀

**Comment by UniqMartin** [comment]
> Sounds like it's worth asking upstream if they aim to support OS X. If not: 💀

I just did in Openwsman/openwsman#89. If we don't hear back from them, I guess it's 💀 time.

**Comment by kkaempf** [comment]
Let me, the openwsman upstream maintainer, answer here directly.

Autotools is deprecated in openwsman because CMake is *much* faster and flexible. However, I will accept patches to the autotools build system ;-)

Adding OS X support to openwsman is fine, if someone sends me a pull request. I myself have no experience with OS X and no way to test/validate a build.

**Comment by UniqMartin** [comment]
> Let me, the openwsman upstream maintainer, answer here directly.

Thanks for chiming in, that's much appreciated! ❤️

> Autotools is deprecated in openwsman because CMake is *much* faster and flexible. However, I will accept patches to the autotools build system ;-)

I don't think it is practical to support multiple build systems in parallel if only one of them is actively maintained by upstream. The Homebrew maintainers certainly won't be able to contribute to keep the autotools build system alive and in good shape, thus I think it's better to get rid of it completely instead of creating the illusion that it will work, but then to get subtly broken builds. It's perfectly fine to only support CMake and we'll happily use that in Homebrew as soon as the current problems are resolved. (To me this sounds more sustainable and that's the point I tried to make in Openwsman/openwsman#88.)

> Adding OS X support to openwsman is fine, if someone sends me a pull request. I myself have no experience with OS X and no way to test/validate a build.

It has been working fine at some point on OS X, thus I don't think there's much to do. I guess there are mainly a few build system issues that need to be ironed out. Ideally, a user of `openwsman` on OS X would step up and help you achieve that. It's perfectly understandable that you can't provide that level of support yourself if you're not an OS X user.

**Comment by apjanke** [comment]
I'm going to take a quick run at this. Identified at least some of the problems for OS X; report is at Openwsman/openwsman#92.

**Comment by MikeMcQuaid** [comment]
@apjanke Any news on this?

**Comment by UniqMartin** [comment]
@MikeMcQuaid I don't think this can be fixed in the short term, thus I filed #2195. We can always bring this back once the build has been properly fixed upstream (if there are any interested OS X users).
